File: web/src/ipc/events.ts
// Rust から届くイベントの名前（`crates/app/src/events.rs`）。
//
// **ここだけが文字列を持ちます。** 型と違って `ts-rs` は定数を書き出さないので、
// 名前は 2 か所にあります。散らさないために、受け取る側はこの定数を使います。

/** メニューが押された。積荷は `AppAction`。 */
export const APP_ACTION = "app:action";

/** 盤面が別のところで変わった。積荷は `Snapshot`。 */
export const BOARD_CHANGED = "board:changed";

/** クイックキャプチャの保存が終わった。積荷は `CaptureResult`。 */
export const CAPTURE_RESULT = "capture:result";

File: crates/app/src/events.rs
//! Rust から webview への一方向のイベント（`docs/DESIGN.md`「コマンドとイベント」）。
//!
//! **3 つだけです。** 増やすほど「いつ何が届くか」を webview 側で数えることに
//! なるので、増やすときは `docs/DESIGN.md` の規則を先に直してください。
//!
//! 名前と積荷だけがここにあり、実際に投げるのは `run.rs` と `ipc.rs`
//! （`tauri::Emitter`）です。1 か所に集めておくのは、コマンドの側が「これは
//! 自分の戻り値で返るのか、イベントで届くのか」を迷わないためです。

use ekanban_core::model::CardId;
use serde::Serialize;
use ts_rs::TS;

/// クイックキャプチャが書いたとき、ほかのウィンドウが盤面を変えたとき。
///
/// 受け取った webview はスナップショットを差し替えます。
pub const BOARD_CHANGED: &str = "board:changed";

/// メニューが押されたとき。webview の dispatcher に流します（`docs/DESIGN.md`「メニューとキー割り当て」）。
pub const APP_ACTION: &str = "app:action";

/// キャプチャの保存が終わったとき。ウィンドウを閉じる / 失敗を出す。
pub const CAPTURE_RESULT: &str = "capture:result";

/// `capture:result` の積荷。
#[derive(Debug, Clone, PartialEq, Eq, Serialize, TS)]
#[serde(
    rename_all = "camelCase",
    rename_all_fields = "camelCase",
    tag = "status"
)]
#[ts(export)]
pub enum CaptureResult {
    /// 書けた。ウィンドウを閉じる。
    Saved { card_id: CardId },
    /// 書けなかった。ウィンドウは開いたまま、理由を出す。
    Failed { detail: String },
}
